analyzers/PapasSim.py

```diff
@@ -103,55 +95,40 @@ def init_display(self):
 
     def process(self, event):
         '''
-           event must contain the
-           
+           event must contain
+             todo once history is implemented
            event will gain
              ecal_clusters:- smeared merged clusters from simulation
              hcal_clusters:- smeared merged clusters from simulation
              tracks:       - tracks from simulation
              baseline_particles:- simulated particles (excluding electrons and muons)
              sim_particles - simulated particles including electrons and muons
-             
-             
         '''
         event.simulator = self 
         if self.is_display:
             self.display.clear()
         pfsim_particles = []
         gen_particles = getattr(event, self.cfg_ana.gen_particles)
         try: 
-            self.simulator.simulate( gen_particles, self.do_reconstruct)
+            self.simulator.simulate( gen_particles)
         except (PropagationError,SimulationError) as err:
             self.mainLogger.error( str(err) + ' -> Event discarded')
             return False
         pfsim_particles = self.simulator.ptcs
-        if self.is_display:
+        if  len(pfsim_particles) == 0 : # deal with case where no particles are produced
```

Comment:
why this? 
if no pfsim_particles, the collection of simparticles in the event must be empty (but it must be there), see below